### DISCUSSION TITLE
Doesn't connecting to database in an edge function make you lose all of its low latency benefits?

### DISCUSSION THREAD

**Discussion by timmolendijk** [supabase/supabase#9529] [Questions]
Hey there, I am trying to understand for which of my use cases I should be considering edge functions over a more traditional setup of running a dedicated single-region virtual server.

On the site [edge functions are being promoted as a convenient way of connecting to your database](https://supabase.com/edge-functions). I'm confused about this selling point, as connecting to a database (that is located in a single fixed region) appears to defeat the purpose of edge functions? Your edge function could be running close to your user, but if it depends on data from a central database you could have just as well hosted your function in the same region as the database.

Am I missing anything here? Does my reasoning make sense?

**Comment by GaryAustin1** [comment]
If all you do is make a call to the database with very little logic around it, then you are correct.  

At some point with multiple database calls per edge call you might get an advantage over doing those same calls from a browser because CORS preflight requests are not needed in server to server calls, but are for each browser call. 

There is also theoretically better data center to data center throughput, but I personally have not seen any info on the infrastructure that Deno and the instances Supabase uses as far as interconnect.
